lib/core-integration/src/Test/Integration/Scenario/API/Wallets.hs

```diff
@@ -1779,3 +1794,39 @@ spec = do
         ru <- request @ApiWallet ctx ("GET", endpoint) Default newName
         expectResponseCode @IO HTTP.status404 ru
         expectErrorMessage (errMsg404NoWallet wid) ru
+
+    describe "WALLETS_RESYNC_01 - \
+        \ force resync eventually get us back to the same point" $ do
+        scenarioWalletResync01 @'Shelley emptyWallet
+        scenarioWalletResync01 @'Byron emptyRandomWallet
+        scenarioWalletResync01 @'Byron emptyIcarusWallet
+
+-- force resync eventually get us back to the same point
+scenarioWalletResync01
+    :: forall style t n wallet.
+        ( n ~ 'Testnet
+        , Discriminate style
+        , HasType (ApiT WalletId) wallet
+        , HasField' "state" wallet (ApiT SyncProgress)
+        , FromJSON wallet
+        , Generic wallet
+        )
+    => (Context t -> IO wallet)
+    -> SpecWith (Context t)
+scenarioWalletResync01 fixture = it "scenario" $ \ctx -> do
+    w <- fixture ctx
+
+    -- 1. Wait for wallet to be synced
+    eventually $ do
+        v <- request @wallet ctx (Link.getWallet @style w) Default Empty
+        verify v [ expectFieldSatisfy @IO #state (== (ApiT Ready)) ]
+
+    -- 2. Force a resync
+    r <- request @wallet ctx
+        (Link.forceResyncWallet @style w) Default Empty
+    verify r [ expectResponseCode @IO HTTP.status204 ]
+
+    -- 3. The wallet eventually re-sync
+    eventually $ do
+        v <- request @wallet ctx (Link.getWallet @style w) Default Empty
+        verify v [ expectFieldSatisfy @IO #state (== (ApiT Ready)) ]
```

Comment:
we are fine here to check only `ApiWallet`'s `state` field, right? If there is no point 2 we are obviously satisfying point 3. It would be nice (although I do not know if it is easy) to show in point 2 upon forcing that `state` is `Syncing ...` and we left `Ready` state after eventually achieving it in point 1

Reply:
> It would be nice (although I do not know if it is easy) to show in point 2 upon forcing that state is Syncing ...

that would be great. +1

Also I think that it would be nice to assert for few other stuff (maybe as separate scenarios):
 - balance (total, available, rewards?)
 - delegation status (if it the wallet was delegating is should be delegating still)
 - addresses (if it makes sense ?)
 - transaction history (whatever it should be after resync)

Reply:
I 100% agree with you guys and I actually thought of this. The main problem I have here is .... race-conditions. 

Technically speaking, right after force-resyncing a wallet, your tip should be at genesis. But, by the time you make the request to produce some assertion... your wallet might as well have restored itself :| ... So, it's kinda hard to come up with reliable assertions that still make it possible to assess the behavior, precisely because this endpoint is meant to be used in situations where something went wrong / wasn't supposed to happen. 

So here, the scenario is much more about exercising the endpoint and making sure that, nothing wrong really happen, whereas the real confirmation that it works come from the logging output.

Reply:
I understand that race conditions are the preventing the addition of the "Syncing" assertion? But checking for the balance, delegation, addresses, transaction history should be doable right? (That would also add confidence that nothing wrong happened with the wallet after resync and it is still usable... perhaps making transaction after _resync_ could be also a test that's nice to have )

Reply:
I've added a few tests but, nothing really satisfactory in terms of testing _states_. If you guys have ideas, I'll happily do this in another PR (merging as is).